src/ocl/convolutionocl.cpp

```diff
@@ -512,23 +524,27 @@ void ConvolutionDescriptor::ConvolutionForward(Handle& handle,
 
     const auto tensors = ConvFwdTensors{xDesc, x, wDesc, w, yDesc, y};
     ValidateTensors(tensors);
-    ValidateAlphaBeta(alpha, beta);
 
     ConvForwardCheckNumerics(handle, tensors, [&]() {
         ValidateGroupCount(xDesc, wDesc, *this);
 
         const auto algorithm_name = AlgorithmName{ConvolutionAlgoToDirectionalString(
             static_cast<miopenConvAlgorithm_t>(algo), conv::Direction::Forward)};
-
-        const auto problem =
-            conv::ProblemDescription{xDesc, wDesc, yDesc, *this, conv::Direction::Forward};
+        Alpha alpha_val(alpha, AlphaBetaStorageDataType(xDesc));
+        Beta beta_val(beta, AlphaBetaStorageDataType(xDesc));
+        const auto problem = conv::ProblemDescription{
```

Comment:
Problem description should also cache the `AlphaBetaCase` enum so that we can use it to quickly make decisions.

Reply:
now initialized in constructor of[ problem_description.hpp](https://github.com/ROCm/MIOpen/blob/83cfe579b5f29c29d5a54202d7f10b86f0370afc/src/include/miopen/conv/problem_description.hpp#L166)